### DISCUSSION TITLE
How to clip ultrasound image

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1174] [Q&A] [answered]
Hi, 
I'm using volume reconstruction module, but the user interface is being reconstructed too, I want to only use the ultrasound image in itself, I've tried using `ClipRectangleOrigin = "200 206" ClipRectangleSize = "532 420" ` in the config file but it does not change anything. 
config file: https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/EYQP5OhNWZBKjt1SGgZAq54B9r0HGn09XN9wj8qE8iEt8w?e=mlAO6i

![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/1b6a65d5-4233-4342-82e0-9ae22d9aeb80)

![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/ba079a43-bc08-4197-a588-a04e5d8e33c8)

I'm using Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB

**Comment by Sunderlandkyl** [comment]
If you are doing the reconstruction in 3D Slicer you need to specify ClipRectangleOrigin and ClipRectangleSize in the device or DataSource

Ex.
https://github.com/PlusToolkit/PlusLibData/blob/d8f1456bd125bcd7e577805cd57fad1ec9cbfb70/ConfigFiles/PlusDeviceSet_Server_EpiphanBiPlane_NDIAurora.xml#L32

**Reply by Sunderlandkyl** [reply]
Also this:

https://github.com/PlusToolkit/PlusLibData/blob/d8f1456bd125bcd7e577805cd57fad1ec9cbfb70/ConfigFiles/BWH/PlusDeviceSet_Server_BKProstate_Epiphan_NDIPolaris_Tamas.xml#L46

**Reply by LeidyMoraV** [reply]
How do I obtain the coordinates? I was trying to use the same as in fcal but it does not change the US image
```
  <Device
      Id="VideoDevice" 
      Type="Epiphan" >
      ClipRectangleOrigin="200 206"
      ClipRectangleSize="532 420">
      <DataSources>
        <DataSource Type="Video" Id="Video" PortUsImageOrientation="MF" ClipRectangleOrigin="263 163 0" ClipRectangleSize="704 598 0" />
      </DataSources>      
      <OutputChannels>
        <OutputChannel Id="VideoStream" VideoDataSourceId="Video" />
      </OutputChannels>
    </Device>
```

I want the image only showing the ultrasound, something like the region in the red plane 
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/7fbb89c3-f604-4971-ac52-2de757019f8b)

**Reply by Sunderlandkyl** [reply]
If you hover over the image in one of the views, you should see some coordinates in the data probe that you can use.

**Reply by LeidyMoraV** [reply]
The superior left point= "694,163 0" , inferior right point="257 608 0"
Then, this means that:
`<DataSource Type="Video" Id="Video" PortUsImageOrientation="MF" ClipRectangleOrigin="257 608 0" ClipRectangleSize="437 445 0" />`
Is this correct? Because I'm not seeing any changes in the US Slicer

**Reply by Sunderlandkyl** [reply]
"437 445 0" is not a valid size, since the third dimension needs to be at least 1 voxel thick, "437 445 1" should be valid.
You can also just specify "437 445" and leave the third dimension off.

**Reply by LeidyMoraV** [reply]
Thank you, I corrected the error but the image from the US is still full size in 3D slicer, what can I do?

**Reply by Sunderlandkyl** [reply]
Can you upload the new config file?

**Reply by LeidyMoraV** [reply]
https://pennstateoffice365-my.sharepoint.com/:u:/g/personal/lfm5705_psu_edu/EbhAoxrv0RFJg-XMh3YFbaQB3pAg7lk8pZ1DF2-aqWk9cQ?e=3sft5q

**Reply by Sunderlandkyl** [reply]
I just reviewed the Slicer scene from yesterday. It looks like the size of your clip region goes out of bounds.

Image: 1024x768
Clip region origin: [257, 608]
Clip region opposite corner: [(257+437), (608+445)] = [695, 1053]

The 1053 goes out of bounds of the original image.

**Reply by LeidyMoraV** [reply]
I see, based on that, should it be something like this?
Coordinates Corner 1 [263 613]
Coordinates Corner 2 [710 160]

Clip Region Origin: [263 613]
Clip Size: [447 -453]

I was thinking the corner points like this, maybe thats my mistake?
![image](https://github.com/PlusToolkit/PlusLib/assets/162748695/36c02512-d337-4e99-a239-51235756f583)

When I used fcal, the interest region was ClipRectangleOrigin="200 206" ClipRectangleSize="532 420"

**The thing is no matter what coordinates I use the US image doesn't change at all.**

**Reply by Sunderlandkyl** [reply]
You can't use negative numbers in ClipRectangleSize, so ClipRectangleOrigin should always be the corner closest to [0,0].

If the fCal clip rectangle works, then I would use that. Otherwise I would start by making the clipping ClipRectangleOrigin="0 0" and change the ClipRectangleOrigin and ClipRectangleSize until you get the rectangle you want.

If the clip rectangle is invalid then the size will not change.

**Reply by LeidyMoraV** [reply]
Even if I use ClipRectangleOrigin="0 0" ClipRectangleSize="50 50" the US image does not change, maybe there is another thing affecting the size?

**Reply by Sunderlandkyl** [reply]
Hmm... not that I can see.
Can you set log level to "Trace", run the server, then click the "Latest log" button and upload that file?

**Reply by LeidyMoraV** [reply]
https://pennstateoffice365-my.sharepoint.com/:t:/g/personal/lfm5705_psu_edu/EVgX_VzSSGdLp-rBykRgIc8BiO8VJVksAeAw3Cx22MYdtg?e=P4yuGP

**Reply by LeidyMoraV** [reply, marked as answer]
Hi, the easiest way to do it it's to use the _Clipping_ tool in the _Volume Reconstruction_ module in Slicer.

**Reply by Sunderlandkyl** [reply]
Great, I'm still not sure why the clipping of the video source itself is not working, but I'm glass you found a solution!

**Reply by LeidyMoraV** [reply]
Hi! In case someone else has the same problem it the future, it seemed like the problem was the fact that I was not able to run everything as an administrator (since I'm using a university's computer), so changes like the region clip were not properly working even though it didn't show any error message.
